### PR TITLE
Artist recommendations 

### PR DESCRIPTION
**Pull request by pepopowitz**
A collaboration with @jonallured.

Addresses https://artsyproduct.atlassian.net/browse/DISCO-870.

This PR hooks up the artist recommendation section for users who have access to a specific lab feature.

We're using a separate QueryRenderer for the ArtistRecommendations component, for now, because it was the easiest way to make Relay not query for that data unless the user has access to the recommendations feature.

Here's what the page looks like for someone with access to the feature: 

image

Not super useful or usable yet, but it emits a label that reads "Related to __artist name__" where the recommendations will go.

To prove that this won't do anything for most users, here are two more screenshots.

1. what the page looks like for someone logged in, but who doesn't have access to the feature:

image

2. what the page looks like for someone not logged in:

image

### CODE REVIEW

File: src/Apps/Artist/ArtistApp.tsx [hunk 1]
```diff
@@ -77,6 +98,7 @@ export const ArtistAppFragmentContainer = createFragmentContainer(ArtistApp, {
       id
       ...ArtistHeader_artist
       ...NavigationTabs_artist
+      ...ArtistRecommendations_artist
```
Comment (mzikherman) [inline src/Apps/Artist/ArtistApp.tsx]: This winds up adding a potentially heavy data load to every request, even if its for an unauthenticated user or a user who doesn't have the lab feature. We can actually prevent against that, and only fetch this fragment if the user has the lab feature.

That concept (does this request need/have access to the recommendations part), can be handled higher up, and here can be referred to as a variable: `...ArtistRecommendations_artist @include(if: $includeRecommendations)`.

For purposes of storybooks and ease of this initial development, you can possibly hard-code that to true (or we can have it read from ENV, or something).

Then, when it comes to actually using it in Force w/ it being actually set properly, you can pass it in via context.

Basically like: https://github.com/artsy/force/blob/203673e3ce5c0c936d51ac3c66e19cbf471e8c70/src/desktop/apps/artist/server.tsx#L33-L36 for the server and https://github.com/artsy/force/blob/203673e3ce5c0c936d51ac3c66e19cbf471e8c70/src/desktop/apps/artist/client.tsx#L13-L16 in the client.

Let me know if you have questions or need help setting that up!

Comment (mzikherman) [reply src/Apps/Artist/ArtistApp.tsx]: Realizing something similar was happening for the 'pricing' lab feature on the artwork page, so just made a similar comment there: https://github.com/artsy/reaction/pull/2222/files#r270593368

Comment (pepopowitz) [reply src/Apps/Artist/ArtistApp.tsx]: 👍 Thank you! I'll work on this today and reach out with questions.

File: src/Apps/Artist/Components/ArtistRecommendations.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+export const ArtistRecommendationsFragmentContainer = createFragmentContainer(
+  ArtistRecommendations,
+  graphql`
+    fragment ArtistRecommendations_artist on Artist {
+      name
+    }
+  `
+)
+
+export const ArtistRecommendationsQueryRenderer: React.FC<{
```
Comment (pepopowitz) [inline src/Apps/Artist/Components/ArtistRecommendations.tsx]: @mzikherman Let us know if this looks different than you'd have expected!

Comment (mzikherman) [reply src/Apps/Artist/Components/ArtistRecommendations.tsx]: Yep looks great!

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 16719dd3ec407bbf3495d6c8e35b8ea8d0f4a084

https://deploy-preview-2236--artsy-reaction.netlify.com

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  ArtistRecommendations,
+  graphql`
+    fragment ArtistRecommendations_artist on Artist {
+      name
+    }
+  `
+)
+
+export const ArtistRecommendationsQueryRenderer: React.FC<{
+  artist: Overview_artist
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: It might be better not to pass the whole artist object here, but just `artistID`. Then the invocation is `<ArtistRecommendations artistID=...`, which might make it easier to reuse this component outside the overview tab.

Although, that might be a premature optimization since I don't believe there's any plans for that, so feel free to disregard.

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+}> = ({ artist }) => {
+  const { relayEnvironment } = useContext(SystemContext)
```
Comment (mzikherman) [inline src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: You should be able to also destructure `user` here as well (and have lab feature access). What do you think about moving the `showRecommendations && ...` logic [from here](https://github.com/artsy/reaction/pull/2236/files#diff-bfc57d8ce9d60a71675e59322d85c6f3R182) into this query renderer so that it's all just in one place (`return null` if the user doesn't have access).

It might be easier to then keep track/change that condition later.

Comment (pepopowitz) [reply src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: Yeah, good call! I will update that.

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx (hunk 3 above)
Comment (pepopowitz) [reply src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: I did feel a little weird about pulling in an object from a different component (`Overview_artist`)...so I'll make this change.

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx (hunk 4 above)
Comment (pepopowitz) [reply src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: Looking into this, I prefer it where it's at. The `<ArtistRecommendations>` component [gets wrapped in a `<Row><Col>`](https://github.com/artsy/reaction/pull/2236/files#diff-bfc57d8ce9d60a71675e59322d85c6f3R182). I'd rather the Row/Col be in the `Overview` than in the `ArtistRecommendations`, and I don't think they should be emitted unless the recommendations are going to be emitted.

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx (hunk 3 above)
Comment (pepopowitz) [reply src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: Done!

File: src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx (hunk 4 above)
Comment (mzikherman) [reply src/Apps/Artist/Routes/Overview/Components/ArtistRecommendations.tsx]: Cool, makes sense!

**Comment by mzikherman** [comment]
Oh yea, looking great! 😎

**Comment by artsyit** [comment]
:rocket: PR was released in v15.11.21 :rocket:
